src/atomic-queue/AtomicQueue.sol

```diff
@@ -202,47 +187,76 @@ contract AtomicQueue is ReentrancyGuard {
-        uint256 assetsForWant;
         for (uint256 i; i < users.length; ++i) {
             AtomicRequest storage request = userAtomicRequest[users[i]][offer][want];
 
             if (request.inSolve) revert AtomicQueue__UserRepeated(users[i]);
             if (block.timestamp > request.deadline) revert AtomicQueue__RequestDeadlineExceeded(users[i]);
             if (request.offerAmount == 0) revert AtomicQueue__ZeroOfferAmount(users[i]);
 
-            // User gets whatever their atomic price * offerAmount is.
-            assetsForWant += _calculateAssetAmount(request.offerAmount, request.atomicPrice, offerDecimals);
-
-            // If all checks above passed, the users request is valid and should be fulfilled.
+            assetsForWant += rate.mulDivDown(request.offerAmount, offerDecimals);
             assetsToOffer += request.offerAmount;
             request.inSolve = true;
-            // Transfer shares from user to solver.
             offer.safeTransferFrom(users[i], solver, request.offerAmount);
         }
+    }
 
-        IAtomicSolver(solver).finishSolve(runData, msg.sender, offer, want, assetsToOffer, assetsForWant);
+    /**
+     * @notice Internal function to handle final phase of solve
+     * @dev Transfers want tokens from solver to users and cleans up state
+     * @dev Uses same NAV rate for all calculations to ensure consistency
+     * @param offer the ERC20 offer token
+     * @param want the ERC20 want token
+     * @param users array of users to process
+     * @param solver the solver address providing want tokens
+     */
+    function _finalizeSolve(ERC20 offer, ERC20 want, address[] calldata users, address solver) internal {
+        uint256 rate = accountant.getRateInQuoteSafe(want);
+        uint256 offerDecimals = 10 ** offer.decimals();
 
         for (uint256 i; i < users.length; ++i) {
             AtomicRequest storage request = userAtomicRequest[users[i]][offer][want];
 
             if (request.inSolve) {
-                // We know that the minimum price and deadline arguments are satisfied since this can only be true if
-                // they were.
-
-                // Send user their share of assets.
-                uint256 assetsToUser = _calculateAssetAmount(request.offerAmount, request.atomicPrice, offerDecimals);
+                uint256 amountOut = rate.mulDivDown(request.offerAmount, offerDecimals);
 
-                want.safeTransferFrom(solver, users[i], assetsToUser);
+                want.safeTransferFrom(solver, users[i], amountOut);
 
                 emit AtomicRequestFulfilled(
-                    users[i], address(offer), address(want), request.offerAmount, assetsToUser, block.timestamp
+                    users[i], address(offer), address(want), request.offerAmount, amountOut, block.timestamp
                 );
 
-                // Set shares to withdraw to 0.
                 request.offerAmount = 0;
                 request.inSolve = false;
```

Comment:
Can you take a look at the code here should we add `request.deadline=0`

Reply:
It seems like this suggestion is more of a nice-to-have improvement rather than a critical fix. It doesn’t cause significant harm but mainly enhances readability.